agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/exporter/Exporter.java

```diff
@@ -766,14 +814,16 @@ private void exportRequest(SpanData span) {
   }
 
   @Nullable
-  private static String getSource(Attributes attributes) {
+  private static String getSource(Attributes attributes, SpanContext spanContext) {
     // this is only used by the 2.x web interop bridge
     // for ThreadContext.getRequestTelemetryContext().getRequestTelemetry().setSource()
     String source = attributes.get(AI_SPAN_SOURCE_KEY);
     if (source != null) {
       return source;
     }
-    source = attributes.get(AI_SPAN_SOURCE_APP_ID_KEY);
+
+    source = spanContext.getTraceState().get("az");
```

Comment:
what is az?

Reply:
legacy appId exchange is done using trace state, "az" was the azure monitor "key" that is used